### PR TITLE
feat: add color map for footprint type color map

### PR DESCRIPTION
**Pull request by mlu-ssoft**
- add color map for footprint type coloMap
- fix error for footprint type classification with only one color

### CODE REVIEW

File: libs/map/ui-map/src/lib/true-color-image-layer/stac/classification.color-map.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  for (let i = 0; i <= options.length; ++i) {
+    const item = options[i];
+    if (item) {
+      const color = hexToRgb(`#${item['color-hint']}`);
+
+      stops.push([['==', ['band', 1], item.value], color ? color : ['color', 0, 0, 0]]);
+    }
+  }
+
+  return stops;
```
Comment (KatarzynaPaszko) [inline libs/map/ui-map/src/lib/true-color-image-layer/stac/classification.color-map.ts:50]:
```suggestion
  const steps: [['==', ['band', number], number], TColorMapPair][] = [];

  for (let i = 0; i <= options.length; ++i) {
    const item = options[i];
    if (item) {
      const color = hexToRgb(`#${item['color-hint']}`);

      steps.push([['==', ['band', 1], item.value], color ? color : ['color', 0, 0, 0]]);
    }
  }

  return steps;
```

File: libs/map/ui-map/src/lib/true-color-image-layer/stac/color-map.color-map.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    z.literal('oxygen'),
+    z.literal('par'),
+    z.literal('phase'),
+    z.literal('salinity'),
+    z.literal('temperature'),
+    z.literal('turbidity'),
+    z.literal('velocity-blue'),
+    z.literal('velocity-green'),
+    z.literal('cubehelix'),
+  ]),
```
Comment (KatarzynaPaszko) [inline libs/map/ui-map/src/lib/true-color-image-layer/stac/color-map.color-map.ts:50]: should we add just string as well, to not get error once there will be some new color added that is not mapped here?

File: libs/map/ui-map/src/lib/true-color-image-layer/stac/color-map.color-map.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    colors.reverse();
+  }
+
+  for (let i = 0; i < steps; i++) {
+    stops[i * 2] = min + i * delta;
+    stops[i * 2 + 1] = colors[i];
+  }
+
+  return stops;
+};
```
Comment (KatarzynaPaszko) [inline libs/map/ui-map/src/lib/true-color-image-layer/stac/color-map.color-map.ts:74]: should it be `stops` or `steps`  ?

File: libs/map/ui-map/src/lib/true-color-image-layer/use-stac-layer.hook.tsx [hunk 4]
```diff
@@ -75,8 +75,11 @@ export const useStacLayer = () => {
           return options;
         },
       });
+      // todo remove after rewriting ol-stac library
+      setTimeout(() => {
+        handleSourceReady();
+      }, 1000);
```
Comment (KatarzynaPaszko) [inline libs/map/ui-map/src/lib/true-color-image-layer/use-stac-layer.hook.tsx:81]: Why it is added ?

File: libs/map/ui-map/src/lib/true-color-image-layer/stac/color-map.color-map.ts (hunk 2 above)
Comment (mlu-ssoft) [reply libs/map/ui-map/src/lib/true-color-image-layer/stac/color-map.color-map.ts:50]: no, library which we are using supports only those algotithms

File: libs/map/ui-map/src/lib/true-color-image-layer/stac/color-map.color-map.ts (hunk 3 above)
Comment (mlu-ssoft) [reply libs/map/ui-map/src/lib/true-color-image-layer/stac/color-map.color-map.ts:74]: it was stops in the examples

File: libs/map/ui-map/src/lib/true-color-image-layer/use-stac-layer.hook.tsx (hunk 4 above)
Comment (mlu-ssoft) [reply libs/map/ui-map/src/lib/true-color-image-layer/use-stac-layer.hook.tsx:81]: it is hack for zooming to Footprint. when data is passwd library is firing event surceready. it is during initialization so when we listening on this event it is already fired and `handleSourceReady` won't be executed. If we add setTimeout then it works properly

File: libs/map/ui-map/src/lib/true-color-image-layer/stac/classification.color-map.ts (hunk 1 above)
Comment (mlu-ssoft) [reply libs/map/ui-map/src/lib/true-color-image-layer/stac/classification.color-map.ts:50]: it is `stops` in the examples

**Review by KatarzynaPaszko — APPROVED** [review]

**Pull request merged**